### PR TITLE
Added secure-http flag, defaults to true

### PR DESCRIPTION
**Pull request by cinamo**
@Seldaek Something like this for issue #4777?

It seems somewhere there's still http://packagist.org links generated, so I added a little fix for those specifically. I guess that shouldn't be necessary, so maybe you can point me to the right spot to fix?

### CODE REVIEW

**Comment by barryvdh** [comment]
You should usually not change multiple things in 1 PR, like unrelated CS fixes. You can do that in a different PR :)

**Comment by cinamo** [comment]
Ouch, you're right of course. Meant to do that for just the method I added. What's the best way to proceed?

File: src/Composer/Util/RemoteFilesystem.php [hunk 1]
```diff
@@ -993,11 +1019,11 @@ private function getCertificateCnAndFp($url, $options)
     private function getUrlAuthority($url)
     {
         $defaultPorts = array(
-            'ftp' => 21,
-            'http' => 80,
-            'https' => 443,
+            'ftp'       => 21,
+            'http'      => 80,
+            'https'     => 443,
```
Comment (stof) [inline src/Composer/Util/RemoteFilesystem.php]: -1 for this CS change. Aligning array arrows makes the maintenance a huge pain by generating extra useless diffs regularly (which means extra useless git conflicts potentially, and also breaks the usage of ``git blame``)

File: src/Composer/Util/RemoteFilesystem.php [hunk 2]
```diff
@@ -248,12 +250,28 @@ protected function get($originUrl, $fileUrl, $additionalOptions = array(), $file
             $fileUrl = 'http://' . gethostbyname('packagist.org') . substr($fileUrl, 20);
         }
 
-        $ctx = StreamContextFactory::getContext($fileUrl, $options, array('notification' => array($this, 'callbackGet')));
+        $ctx = StreamContextFactory::getContext($fileUrl, $options,
+            array('notification' => array($this, 'callbackGet')));
```
Comment (stof) [inline src/Composer/Util/RemoteFilesystem.php]: such wrapping does not make sense (and is not PSR-2 compliant btw)

**Comment by curry684** [comment]
Wtf is happening in Travis there, did Pear get nuked?

**Comment by Seldaek** [comment]
Thanks!

**Pull request merged**